shell/platform/android/io/flutter/plugin/editing/InputConnectionAdaptor.java

```diff
@@ -133,6 +136,25 @@ public boolean setComposingText(CharSequence text, int newCursorPosition) {
         return result;
     }
 
+    @Override
+    @TargetApi(21)
+    public boolean finishComposingText() {
```

Comment:
Is this API only called on SDK 21?

Reply:
No, but the updateCursorAnchorInfo call is 21+ only, I think the affected devices that need the fix are all 21+. I can add an additional check to only do this for Samsung if that is something you think that is needed.

Reply:
OK, thanks! I think it's fine to call the cursor update on non-Samsung devices, but this method probably shouldn't be annotated with `@TargetApi(21)` since it's called on lower APIs. If the tests are running it SDK 16 and crashing you can update them to run it on sdk 27 with the `@Config` tag (`SingleViewPresentationTest.java` is an example). I think how it's written right now the tests may just be skipping this every time since by default they're run on SDK 16.